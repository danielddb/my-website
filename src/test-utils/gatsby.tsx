import { FixedObject } from 'gatsby-image';

export const createGatsbyFixedImageMock = (): FixedObject => ({
  base64:
    'data:image/jpeg;base64,/9j/2wBDABALDA4MChAODQ4SERATGCgaGBYWGDEjJR0oOjM9PDkzODdASFxOQERXRTc4UG1RV19iZ2hnPk1xeXBkeFxlZ2P/2wBDARESEhgVGC8aGi9jQjhCY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2P/wgARCAAUABQDASIAAhEBAxEB/8QAGAABAAMBAAAAAAAAAAAAAAAAAAECBAX/xAAXAQEBAQEAAAAAAAAAAAAAAAABAwAC/9oADAMBAAIQAxAAAAHPVWbVIOrmLbUHn//EABwQAAICAgMAAAAAAAAAAAAAAAABAhEDEBIhMf/aAAgBAQABBQKy6b94tbydmGCHBM//xAAUEQEAAAAAAAAAAAAAAAAAAAAg/9oACAEDAQE/AR//xAAVEQEBAAAAAAAAAAAAAAAAAAABEP/aAAgBAgEBPwEjP//EABoQAAIDAQEAAAAAAAAAAAAAAAARARAxIVL/2gAIAQEABj8CUSj0cZlYK//EABwQAAIDAAMBAAAAAAAAAAAAAAABESExUWFxkf/aAAgBAQABPyHkexBVwy2JbD0c5vD+HUIh8hFdg6m2pgts8P/aAAwDAQACAAMAAAAQrAeA/8QAGBEAAwEBAAAAAAAAAAAAAAAAAAExEBH/2gAIAQMBAT8QqOijz//EABcRAQEBAQAAAAAAAAAAAAAAAAEAETH/2gAIAQIBAT8QOFjdEht//8QAHhABAQACAgIDAAAAAAAAAAAAAREAITFBUWFxgaH/2gAIAQEAAT8QlUAsjTlUNNinPd8mSN7O+H5hwCbq2jEs+VNXA+J9QcoDUX7856WQ8YuQ6yOGf//Z',
  width: 64,
  height: 64,
  src: '/static/6f6ccd0263b69cfd693f71c17be8ac7c/cf62c/avatar.jpg',
  srcSet:
    '/static/6f6ccd0263b69cfd693f71c17be8ac7c/cf62c/avatar.jpg 1x,\n/static/6f6ccd0263b69cfd693f71c17be8ac7c/eee8e/avatar.jpg 1.5x,\n/static/6f6ccd0263b69cfd693f71c17be8ac7c/6dc0c/avatar.jpg 2x'
});
