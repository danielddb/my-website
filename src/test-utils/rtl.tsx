import { render, RenderOptions } from '@testing-library/react';
import React from 'react';
import { ThemeProvider } from 'styled-components';
import light from '../themes/light';

const Wrapper = ({ children }) => (
  <ThemeProvider theme={light}>{children}</ThemeProvider>
);

const customRender = (
  ui: React.ReactElement,
  options?: Omit<RenderOptions, 'queries'>
) => render(ui, { wrapper: Wrapper, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
