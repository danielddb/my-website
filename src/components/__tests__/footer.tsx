import { useStaticQuery } from 'gatsby';
import { axe } from 'jest-axe';
import React from 'react';
import { render } from 'test-utils/rtl';
import Footer from '../footer';

const mockStaticQuery = useStaticQuery as jest.Mock<any>;

beforeEach(() => {
  mockStaticQuery.mockImplementation(() => ({
    site: {
      siteMetadata: {
        socials: [
          {
            name: 'GitHub',
            url: 'https://github.com/danielddb',
            icon: 'github'
          },
          {
            name: 'CodeSandbox',
            url: 'https://codesandbox.io/u/danielddb',
            icon: 'code-sandbox'
          },
          {
            name: 'LinkedIn',
            url: 'https://www.linkedin.com/in/dandawsonbrown',
            icon: 'linkedin'
          }
        ]
      }
    }
  }));
});

test('it should render correctly', () => {
  const { container } = render(<Footer />);

  expect(container).toMatchSnapshot();
});

test('it should render the correct nav links', () => {
  const { getByText, getByTitle } = render(<Footer />);

  expect(getByText(/blog/i)).toBeInTheDocument();
  expect(getByTitle(/github/i)).toBeInTheDocument();
  expect(getByTitle(/linkedin/i)).toBeInTheDocument();
  expect(getByTitle(/codesandbox/i)).toBeInTheDocument();
});

test('it should render with no accessibility issues', async () => {
  const { container } = render(<Footer />);
  const results = await axe(container);
  expect(results).toHaveNoViolations();
});
