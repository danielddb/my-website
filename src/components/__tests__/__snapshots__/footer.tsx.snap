// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should render correctly 1`] = `
.c1 {
  margin: 0 auto;
  padding: 0 1.5rem;
}

.c1 > :first-child {
  margin-top: 0;
}

.c4 {
  height: 4rem;
}

.c3 {
  -webkit-text-decoration: none;
  text-decoration: none;
  display: inline-block;
}

.c3:hover,
.c3:focus {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c6 {
  fill: #3701b3;
  -webkit-transition: -webkit-transform 500ms ease-in-out;
  -webkit-transition: transform 500ms ease-in-out;
  transition: transform 500ms ease-in-out;
  -webkit-transform-origin: 50% 50%;
  -ms-transform-origin: 50% 50%;
  transform-origin: 50% 50%;
}

.c2:focus .c6,
.c2:hover .c6 {
  -webkit-transform: rotate(360deg);
  -ms-transform: rotate(360deg);
  transform: rotate(360deg);
}

.c5 {
  fill: #101010;
  -webkit-transition: -webkit-transform 500ms ease-in-out;
  -webkit-transition: transform 500ms ease-in-out;
  transition: transform 500ms ease-in-out;
  -webkit-transform-origin: 50% 50%;
  -ms-transform-origin: 50% 50%;
  transform-origin: 50% 50%;
  -webkit-transform: scale(0.8);
  -ms-transform: scale(0.8);
  transform: scale(0.8);
}

.c2:focus .c5,
.c2:hover .c5 {
  -webkit-transform: scale(0.6);
  -ms-transform: scale(0.6);
  transform: scale(0.6);
}

.c7 {
  padding: 2rem 0;
}

.c7 > :first-child {
  margin-top: 0;
}

.c13 {
  margin: 0;
  font-size: 1.125rem;
  line-height: 1rem;
  color: #888888;
}

.c12 {
  display: inline-block;
  vertical-align: middle;
  width: 1.5rem;
  height: 1.5rem;
  fill: currentColor;
}

.c8 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 0 0 0 -2rem;
  padding: 0;
  list-style: none;
}

.c9 {
  line-height: 1rem;
  padding-left: 2rem;
}

.c11 {
  -webkit-appearance: none;
  border: none;
  background: transparent;
  margin: 0;
  padding: 0;
  color: #3701b3;
}

.c11:hover,
.c11:focus {
  color: #4b1abb;
}

.c11:active {
  color: #3201a1;
}

.c11:disabled {
  color: #888888;
}

.c10 {
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: border-bottom 300ms ease-in-out;
  transition: border-bottom 300ms ease-in-out;
}

.c10:hover,
.c10:focus {
  -webkit-text-decoration: none;
  text-decoration: none;
  box-shadow: 0 3px 0 0 #4b1abb;
}

.c0 {
  padding: 3rem 0;
  border-top: 1px solid #e7e7e7;
  text-align: center;
}

.c0 > :first-child {
  margin-top: 0;
}

@media (min-width:36em) {
  .c1 {
    max-width: 36em;
  }
}

<div>
  <div
    class="sc-AxheI c0"
    spacing="3"
  >
    <div
      class="c1"
    >
      <a
        class="c2 c3"
        href="/"
      >
        <svg
          aria-labelledby="footer-logo"
          class="c4"
          overflow="visible"
          viewBox="-4 -4.812 320 360"
          xmlns="http://www.w3.org/2000/svg"
        >
          <title
            id="footer-logo"
          >
            Dan Dawson-Brown Logo
          </title>
          <path
            class="c5"
            d="M255.944 85.702h-98.782l.112.053c-.954-.019-1.898-.053-2.864-.053H90.92v148.755l-34.863 30.216H154.41c56.243 0 92.237-33.541 92.237-89.86 0-26.131-7.76-47.185-21.683-62.261l30.98-26.85zM152.66 227.126h-17.998V123.249h17.998c29.497 0 48.744 17.521 48.744 51.563 0 34.292-18.497 52.314-48.744 52.314z"
          />
          <path
            class="c6"
            d="M156 350.375c-4.661 0-9.156-1.107-12.656-3.119l-130.66-75.092C5.453 268.008 0 258.586 0 250.244V100.129c0-8.34 5.453-17.762 12.684-21.918l130.66-75.092C146.844 1.107 151.338 0 156 0c4.661 0 9.156 1.107 12.656 3.119l130.66 75.092C306.547 82.367 312 91.789 312 100.129v150.115c0 8.342-5.453 17.764-12.684 21.92l-130.66 75.092c-3.5 2.012-7.995 3.119-12.656 3.119zM156 16c-2.182 0-3.85.512-4.684.992l-130.66 75.09C18.35 93.408 16 97.469 16 100.129v150.115c0 2.66 2.35 6.721 4.656 8.047l130.66 75.092c.834.48 2.502.992 4.684.992s3.85-.512 4.684-.992l130.66-75.092c2.307-1.326 4.656-5.387 4.656-8.047V100.129c0-2.66-2.35-6.721-4.656-8.047l-130.66-75.09c-.834-.48-2.503-.992-4.684-.992z"
          />
        </svg>
      </a>
      <div
        class="c7"
        spacing="2"
      >
        <ul
          class="c8"
        >
          <li
            class="c9"
          >
            <a
              class="c10"
              href="/blog"
            >
              Blog
            </a>
          </li>
          <li
            class="c9"
          >
            <a
              aria-label="Dan Dawson-Brown's GitHub Page"
              class="c11"
              href="https://github.com/danielddb"
              rel="noopener"
              target="_blank"
            >
              <svg
                aria-labelledby="navbar-github"
                class="c12"
                viewBox="0 0 438.549 438.549"
                xmlns="http://www.w3.org/2000/svg"
              >
                <title
                  id="navbar-github"
                >
                  Github Icon
                </title>
                <path
                  d="M409.132 114.573c-19.608-33.596-46.205-60.194-79.798-79.8-33.598-19.607-70.277-29.408-110.063-29.408-39.781 0-76.472 9.804-110.063 29.408-33.596 19.605-60.192 46.204-79.8 79.8C9.803 148.168 0 184.854 0 224.63c0 47.78 13.94 90.745 41.827 128.906 27.884 38.164 63.906 64.572 108.063 79.227 5.14.954 8.945.283 11.419-1.996 2.475-2.282 3.711-5.14 3.711-8.562 0-.571-.049-5.708-.144-15.417a2549.81 2549.81 0 01-.144-25.406l-6.567 1.136c-4.187.767-9.469 1.092-15.846 1-6.374-.089-12.991-.757-19.842-1.999-6.854-1.231-13.229-4.086-19.13-8.559-5.898-4.473-10.085-10.328-12.56-17.556l-2.855-6.57c-1.903-4.374-4.899-9.233-8.992-14.559-4.093-5.331-8.232-8.945-12.419-10.848l-1.999-1.431c-1.332-.951-2.568-2.098-3.711-3.429-1.142-1.331-1.997-2.663-2.568-3.997-.572-1.335-.098-2.43 1.427-3.289 1.525-.859 4.281-1.276 8.28-1.276l5.708.853c3.807.763 8.516 3.042 14.133 6.851 5.614 3.806 10.229 8.754 13.846 14.842 4.38 7.806 9.657 13.754 15.846 17.847 6.184 4.093 12.419 6.136 18.699 6.136 6.28 0 11.704-.476 16.274-1.423 4.565-.952 8.848-2.383 12.847-4.285 1.713-12.758 6.377-22.559 13.988-29.41-10.848-1.14-20.601-2.857-29.264-5.14-8.658-2.286-17.605-5.996-26.835-11.14-9.235-5.137-16.896-11.516-22.985-19.126-6.09-7.614-11.088-17.61-14.987-29.979-3.901-12.374-5.852-26.648-5.852-42.826 0-23.035 7.52-42.637 22.557-58.817-7.044-17.318-6.379-36.732 1.997-58.24 5.52-1.715 13.706-.428 24.554 3.853 10.85 4.283 18.794 7.952 23.84 10.994 5.046 3.041 9.089 5.618 12.135 7.708 17.705-4.947 35.976-7.421 54.818-7.421s37.117 2.474 54.823 7.421l10.849-6.849c7.419-4.57 16.18-8.758 26.262-12.565 10.088-3.805 17.802-4.853 23.134-3.138 8.562 21.509 9.325 40.922 2.279 58.24 15.036 16.18 22.559 35.787 22.559 58.817 0 16.178-1.958 30.497-5.853 42.966-3.9 12.471-8.941 22.457-15.125 29.979-6.191 7.521-13.901 13.85-23.131 18.986-9.232 5.14-18.182 8.85-26.84 11.136-8.662 2.286-18.415 4.004-29.263 5.146 9.894 8.562 14.842 22.077 14.842 40.539v60.237c0 3.422 1.19 6.279 3.572 8.562 2.379 2.279 6.136 2.95 11.276 1.995 44.163-14.653 80.185-41.062 108.068-79.226 27.88-38.161 41.825-81.126 41.825-128.906-.01-39.771-9.818-76.454-29.414-110.049z"
                />
              </svg>
            </a>
          </li>
          <li
            class="c9"
          >
            <a
              aria-label="Dan Dawson-Brown's CodeSandbox Page"
              class="c11"
              href="https://codesandbox.io/u/danielddb"
              rel="noopener"
              target="_blank"
            >
              <svg
                aria-labelledby="navbar-code-sandbox"
                class="c12"
                preserveAspectRatio="xMidYMid"
                viewBox="0 0 256 296"
                xmlns="http://www.w3.org/2000/svg"
              >
                <title
                  id="navbar-code-sandbox"
                >
                  CodeSandbox Icon
                </title>
                <path
                  d="M115.498 261.088v-106.61L23.814 101.73v60.773l41.996 24.347v45.7l49.688 28.54zm23.814.627l50.605-29.151V185.78l42.269-24.495v-60.011l-92.874 53.621v106.82zm80.66-180.887l-48.817-28.289-42.863 24.872-43.188-24.897-49.252 28.667 91.914 52.882 92.206-53.235zM0 222.212V74.495L127.987 0 256 74.182v147.797l-128.016 73.744L0 222.212z"
                />
              </svg>
            </a>
          </li>
          <li
            class="c9"
          >
            <a
              aria-label="Dan Dawson-Brown's LinkedIn Page"
              class="c11"
              href="https://www.linkedin.com/in/dandawsonbrown"
              rel="noopener"
              target="_blank"
            >
              <svg
                aria-labelledby="navbar-linkedin"
                class="c12"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <title
                  id="navbar-linkedin"
                >
                  LinkedIn Icon
                </title>
                <path
                  d="M437 0H75C33.648 0 0 33.648 0 75v362c0 41.352 33.648 75 75 75h362c41.352 0 75-33.648 75-75V75c0-41.352-33.648-75-75-75zM181 406h-60V196h60zm0-240h-60v-60h60zm210 240h-60V286c0-16.54-13.46-30-30-30s-30 13.46-30 30v120h-60V196h60v11.309C286.719 202.422 296.93 196 316 196c40.691.043 75 36.547 75 79.688zm0 0"
                />
              </svg>
            </a>
          </li>
        </ul>
      </div>
      <p
        class="c13"
      >
        Â© Dan Dawson-Brown 
        2020
      </p>
    </div>
  </div>
</div>
`;
